File: ios/Sources/TaigiKeyboard/NextWord/RawNextWordPrediction.swift
import Foundation

// MARK: - Shared-Core Candidate

// Pure logic, Foundation-only. Eligible for cross-platform extraction.

/// Engine-side NextWord prediction row at the service boundary.
///
/// Replaces the previous platform-bound `NextWordService.Prediction` so
/// pure engine code (`NextWordEngine`) can consume prediction results
/// without depending on the service type.
///
/// **Design divergence from `nextword-engine-boundary.md` §2.4**: the
/// original sketch proposed `count: Int` + `source: Source (.user | .dict)`
/// so shared-core ports could re-run `NextWordScorer` themselves. Current
/// scoring happens platform-side inside `NextWordService` and already
/// *merges* dict + user entries with summed scores — splitting that back
/// into `count`/`source` would be lossy (a merged row has no single
/// source). Phase I keeps `score: Double` as an opaque ordering weight;
/// elevating scoring into shared-core is a Phase IV-B follow-up tracked
/// on the shared-core roadmap.
struct RawNextWordPrediction: Equatable {
    let hanzi: String
    let tl: String
    let score: Double
}

File: android/app/src/main/java/com/siansiansu/taigikeyboard/ime/core/nextword/RawNextWordPrediction.kt
package com.siansiansu.taigikeyboard.ime.core.nextword

// region Shared-Core Candidate
// Pure logic, Kotlin stdlib only. Eligible for cross-platform extraction.
// endregion

/**
 * Engine-side NextWord prediction row at the service boundary.
 *
 * Mirrors iOS `NextWord/RawNextWordPrediction.swift`. Replaces the
 * platform-bound `NextWordService.Prediction` so pure engine code
 * (`NextWordEngine`) can consume prediction results without depending on
 * the service type. `score` is an opaque ordering weight — scoring lives
 * inside `NextWordService` (Phase IV-B re-hoists it into shared-core per
 * `nextword-engine-boundary.md` §2.4 impl-revision note).
 */
data class RawNextWordPrediction(
    val hanzi: String,
    val tl: String,
    val score: Double,
)
